src/librustc/hir/lowering.rs

```diff
@@ -3286,7 +3290,7 @@ impl<'a> LoweringContext<'a> {
                     span,
                     "expected 'implicit elided lifetime not allowed' error",
                 );
-                let id = self.sess.next_node_id();
+                let id = self.resolver.next_node_id();
```

Comment:
Could we add a method `next_node_id` on `LoweringContext` that just delegates to `self.resolver` so that the diff becomes smaller in case we want to move it again some time?

Reply:
I personally find such methods pretty painful to deal with as it makes figuring out where things are coming from much harder in my experience, but don't necessarily mind making this change. I doubt we'll be moving it again in the near future, so that seems like an odd argument (we can likely pick a canonical location in this PR). There are also not that many uses.